cro-clib/examples/tx.c

```diff
@@ -0,0 +1,80 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wchar.h>
+#include <assert.h>
+#include "../chain-core.h"
+#include "../chain.h"
+
+void print_address(CroAddressPtr a);
+void print_hex(char *tmp, int length)
+{
+    int i;
+    for (i = 0; i < length; i++)
+    {
+        printf("%X", (unsigned char)tmp[i]);
+    }
+    printf("\n");
+}
+int test_tx()
+{
+    const char *mnemonics = "annual dinosaur deliver hour loop food buddy lift alert obvious thank scorpion young amused climb defy erode blur drip gun require clerk beef armed";
+
+    CroHDWalletPtr hdwallet = NULL;
+    CroAddressPtr staking = NULL;
+    CroAddressPtr staking2 = NULL;
+    CroAddressPtr transfer = NULL;
+    CroAddressPtr transfer2 = NULL;
+    CroAddressPtr viewkey = NULL;
+    CroAddressPtr viewkey2 = NULL;
+    CroFeePtr fee = NULL;
+    cro_create_fee(&fee, "5", "1.0");
+
+    cro_restore_hdwallet(mnemonics, &hdwallet);
+    cro_create_staking_address(hdwallet, Devnet, &staking, 0);
+    cro_create_staking_address(hdwallet, Devnet, &staking2, 1);
+    cro_create_transfer_address(hdwallet, Devnet, &transfer, 0);
+    cro_create_transfer_address(hdwallet, Devnet, &transfer2, 1);
+    cro_create_viewkey(hdwallet, Devnet, &viewkey, 0);
+    cro_create_viewkey(hdwallet, Devnet, &viewkey2, 1);
+    print_address(staking);
+    print_address(staking2);
+    print_address(transfer);
+    print_address(transfer2);
+    print_address(viewkey);
+    print_address(viewkey2);
+
+    CroTxPtr tx = NULL;
+    cro_create_tx(&tx);
+    // compose tx
+    cro_tx_add_txin(tx, "1483dadcae2e6e44bc3248623c7dce749f08270dc9967c6a3ba57f1ce4416b29", 0, "dcro14yj2fc3u3ssvlsfdrr6fu0qmvjx97d405jw9jnwe50ge4y6v53hqr20xzd",
```

Comment:
I think this may be fine -- @calvinaco may comment how he saw it other APIs / bindings.
but it may also be good to add `_raw(...)` functions that, instead of taking these textual representations (hex, bech32, ...), would take directly the byte arrays -- as many of them are 32 bytes (txid, transfer address...), they may be wrapped in some opaque types

Reply:
agree, i will change to raw bytes